sdk/metric/view/view.go

```diff
@@ -17,22 +17,195 @@
+// * Customize which attribute(s) are to be reported by the Instrument.
+//
+// An empty config will match all instruments, and do no modifications.
+type View struct {
+	instrumentName *regexp.Regexp
+	hasWildcard    bool
+	scope          instrumentation.Library
+
+	filterKeys  map[attribute.Key]struct{}
+	viewName    string
+	description string
+	// TODO: Aggregation selection
+}
+
+// New returns a new configured view Config.
+func New(opts ...Option) (View, error) {
+	v := View{
+		filterKeys: map[attribute.Key]struct{}{},
+	}
+
+	for _, opt := range opts {
+		v = opt.apply(v)
+	}
+
+	emptyLibrary := instrumentation.Library{}
+	if v.instrumentName == nil &&
+		v.scope == emptyLibrary {
+		return View{}, fmt.Errorf("must provide at least 1 match option")
+	}
+
+	if v.hasWildcard && v.viewName != "" {
+		return View{}, fmt.Errorf("can not use view name with a wildcard match")
+	}
+
+	return v, nil
+}
+
+// TransformInstrument will check if an instrument description matches this view
+// and will convert it if it does.
+func (v View) TransformInstrument(desc Instrument) (outputDesc Instrument, match bool) {
+	if !v.matchName(desc.InstrumentName) ||
+		!v.matchLibraryName(desc.Scope.Name) ||
+		!v.matchLibrarySchemaURL(desc.Scope.SchemaURL) ||
+		!v.matchLibraryVersion(desc.Scope.Version) {
+		return Instrument{}, false
+	}
+	if v.viewName != "" {
+		desc.InstrumentName = v.viewName
+	}
+	if v.description != "" {
+		desc.InstrumentDescription = v.description
+	}
+	return desc, true
+}
+
+// TransformAttributes filters an attribute set to the keys in the config. If no
+// filter was provided the original set is returned.
+func (v View) TransformAttributes(input attribute.Set) attribute.Set {
+	if len(v.filterKeys) == 0 {
+		return input
+	}
+	out, _ := input.Filter(attribute.Filter(func(kv attribute.KeyValue) bool {
+		_, ok := v.filterKeys[kv.Key]
+		return ok
+	}))
+	return out
+}
+
+// TODO: Provide Transform* for AggregationKind (#2816)
+
+func (v View) matchName(name string) bool {
+	return v.instrumentName == nil || v.instrumentName.MatchString(name)
 }
 
-// TODO (#2837): add getter functions for all the internal fields of a Config.
+func (v View) matchLibraryName(name string) bool {
+	return v.scope.Name == "" || name == v.scope.Name
+}
+
+func (v View) matchLibraryVersion(version string) bool {
+	return v.scope.Version == "" || version == v.scope.Version
+}
+
+func (v View) matchLibrarySchemaURL(schemaURL string) bool {
+	return v.scope.SchemaURL == "" || schemaURL == v.scope.SchemaURL
+}
 
-// Option applies a configuration option value to a view Config.
+// Option applies a Configuration option value to a view Config. All options
+// will be used together to determine match and transforms.
 type Option interface {
-	apply(Config) Config
+	apply(View) View
+}
+
+type optionFunc func(View) View
+
+func (f optionFunc) apply(v View) View {
+	return f(v)
+}
+
+// MatchInstrumentName will match an instrument based on the instruments name.
```

Comment:
```suggestion
// MatchInstrumentName will match an instrument based on the instrument's name.
```

or, to avoid the single quote charater

```suggestion
// MatchInstrumentName will match an instrument based on the its name.
```